src/React.purs

```diff
@@ -463,3 +466,10 @@ type ContextConsumer a = ReactClass { children :: a -> ReactElement }
 
 -- | Create a new context provider/consumer pair given a default value.
 foreign import createContext :: forall a. a -> Context a
+
+-- | Create an element from a function using Hook.
+foreign import createElementHook
+  :: forall props
+   . ({ | props } -> Hook ReactElement)
+  -> { | props }
+  -> ReactElement
```

Comment:
`createHookElement` like `createLeafElement`?

Reply:
Since this all uses `createElement` under the hood, this isn't parametric due to `key` and `ref` props.

Reply:
Sounds good for `createHookElement`. Noted on `key` and `ref`.